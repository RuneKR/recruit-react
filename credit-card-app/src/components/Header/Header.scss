@import "src/colors.scss";

.header-title {
  color: $asb-yellow;
  text-align: center;

  @media only screen and (max-width: 600px) {
    position: absolute;
    right: 20px;
  }
}

.header-menu {
  .ant-menu {
    border-right: none;

    .ant-menu-item {
      padding: 0;
    }
  }
}

.header-menuButton {
  position: absolute;
  top: 20px;
  left: 20px;
  color: white;
  font-size: 25px;
}

.header-menuClose {
  float: right;
  margin-top: 3px;

  &:hover {
    cursor: pointer;
  }
}
