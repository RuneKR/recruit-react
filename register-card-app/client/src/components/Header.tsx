import React from "react";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import { AppBar, Toolbar, IconButton, Typography } from "@material-ui/core";
import MenuIcon from "@material-ui/icons/Menu";
import ArrowBackIcon from "@material-ui/icons/ArrowBack";

export interface HeaderProps {
  title: string;
  isMenuShowing?: boolean;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    menuButton: {
      marginRight: theme.spacing(2),
    },
    title: {
      flexGrow: 1,
      textAlign: 'center'
    },
  })
);

export default function Header({
  title = "Menu",
  isMenuShowing = false,
  ...htmlProps
}: HeaderProps & React.HTMLAttributes<HTMLElement>) {
  const classes = useStyles();

  const icon = isMenuShowing ? (
    <ArrowBackIcon data-testid="ArrowBackIcon" />
  ) : (
    <MenuIcon data-testid="MenuIcon" />
  );

  return (
    <div className={classes.root} {...htmlProps}>
      <AppBar position="static">
        <Toolbar>
          <IconButton
            edge="start"
            className={classes.menuButton}
            color="inherit"
            aria-label="menu"
          >
            {icon}
          </IconButton>
          <Typography variant="h6" className={classes.title}>
            {title}
          </Typography>
        </Toolbar>
      </AppBar>
    </div>
  );
}
