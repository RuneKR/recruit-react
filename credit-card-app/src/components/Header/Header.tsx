import React, { Component } from "react";
import { Layout, Drawer, Icon } from "antd";

import "./Header.scss";
import Menu from "../Menu/Menu";

class Header extends Component {
  state = { isMenuVisible: false };

  showMenu = () => {
    this.setState({
      isMenuVisible: true
    });
  };

  hideMenu = () => {
    this.setState({
      isMenuVisible: false
    });
  };

  render() {
    const title = (
      <div>
        Menu
        <Icon
          type="arrow-left"
          className="header-menuClose"
          onClick={this.hideMenu}
        />
      </div>
    );

    return (
      <Layout.Header>
        <Drawer
          title={title}
          placement="left"
          closable={false}
          onClose={this.hideMenu}
          visible={this.state.isMenuVisible}
        >
          <div className="header-menu">
            <Menu />
          </div>
        </Drawer>
        <Icon
          className="header-menuButton"
          onClick={this.showMenu}
          type="menu-unfold"
        />
        <h2 className="header-title">Register card form</h2>
      </Layout.Header>
    );
  }
}

export default Header;
