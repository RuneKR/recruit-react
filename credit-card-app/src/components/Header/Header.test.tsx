import React from "react";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Header from "./Header";

Enzyme.configure({ adapter: new Adapter() });

describe("Header Component", () => {
  test("header rendered correctly", () => {
    const component = renderer.create(<Header />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("show menu", () => {
    /* 
    Should really be testing for the visibility prop of <Drawer> - but apparently enzyme cannot fine elements 
    rendered outside of render tree https://github.com/airbnb/enzyme/issues/1697. 
    So testing for components' state change instead. I guess needs to be further covered in e2e?
  */

    const component = shallow(<Header />);

    component.find("Icon").simulate("click");
    expect(component.state("isMenuVisible")).toEqual(true);
  });
});
