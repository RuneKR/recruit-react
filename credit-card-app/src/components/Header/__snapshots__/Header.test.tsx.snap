// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Header rendered correctly 1`] = `
<header
  className="ant-layout-header"
>
  <i
    aria-label="icon: menu-unfold"
    className="anticon anticon-menu-unfold header-menuButton"
    onClick={[Function]}
    tabIndex={-1}
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="menu-unfold"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM142.4 642.1L298.7 519a8.84 8.84 0 0 0 0-13.9L142.4 381.9c-5.8-4.6-14.4-.5-14.4 6.9v246.3a8.9 8.9 0 0 0 14.4 7z"
      />
    </svg>
  </i>
  <h2
    className="header-title"
  >
    Register card form
  </h2>
</header>
`;

exports[`Header Component header rendered correctly 1`] = `
<header
  className="ant-layout-header"
>
  <i
    aria-label="icon: menu-unfold"
    className="anticon anticon-menu-unfold header-menuButton"
    onClick={[Function]}
    tabIndex={-1}
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="menu-unfold"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM142.4 642.1L298.7 519a8.84 8.84 0 0 0 0-13.9L142.4 381.9c-5.8-4.6-14.4-.5-14.4 6.9v246.3a8.9 8.9 0 0 0 14.4 7z"
      />
    </svg>
  </i>
  <h2
    className="header-title"
  >
    Register card form
  </h2>
</header>
`;

exports[`Header rendered correctly 1`] = `
<header
  className="ant-layout-header"
>
  <i
    aria-label="icon: menu-unfold"
    className="anticon anticon-menu-unfold header-menuButton"
    onClick={[Function]}
    tabIndex={-1}
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="menu-unfold"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM142.4 642.1L298.7 519a8.84 8.84 0 0 0 0-13.9L142.4 381.9c-5.8-4.6-14.4-.5-14.4 6.9v246.3a8.9 8.9 0 0 0 14.4 7z"
      />
    </svg>
  </i>
  <h2
    className="header-title"
  >
    Register card form
  </h2>
</header>
`;
